### PR TITLE
Fix travis (coding style)

### PR DESCRIPTION
**Pull request by deguif**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -86,7 +86,7 @@ lint:
 .PHONY: check-style
 check-style:
 	docker build -t ruflin/elastica-dev-base-linter -f env/elastica/Lint env/elastica/
-	${RUN_ENV_LINT} php-cs-fixer fix --allow-risky=yes --dry-run
+	${RUN_ENV_LINT} php-cs-fixer fix --allow-risky=yes --dry-run --diff
```
Comment (deguif) [inline Makefile]: Add the `--diff` option to better see why it fails in Travis.

**Comment by ruflin** [comment]
@deguif Is this an automated clean up done by the linting? Do we need any changes to the linting config?

**Comment by deguif** [comment]
The linting changed because of a new PHP-CS-FIXER version and new rules for @Symfony conventions ;)

**Comment by deguif** [comment]
I think that can be merged unless we don't want to follow @Symfony conventions

File: lib/Elastica/Index/Settings.php [hunk 2]
```diff
@@ -281,10 +259,7 @@ public function getRefreshInterval(): string
      *
      * To have this changes made the index has to be closed and reopened
      *
-     * @param string $key   Merge policy key (for ex. expunge_deletes_allowed)
```
Comment (ruflin) [inline lib/Elastica/Index/Settings.php]: I like that it keeps the ones with actual documentation.

**Comment by ruflin** [comment]
Thanks @deguif for the clean up. We now probably have to rebase all PR's.

@thePanz I think we should have a fixed version of php-cs-fixer in our CI build so these kind of things don't happen in the future: https://github.com/ruflin/Elastica/blob/master/env/elastica/Lint#L36 A CI build should be reproducible. WDYT?

**Comment by thePanz** [comment]
Yes, I totally agree @ruflin 
There are 2 ways to accomplish it (IMHO):
- add php-cs-fixer as a dev dependency to composer.json
- use Phive and have it pulled in with a fixed version/release

File: lib/Elastica/Exception/Connection/GuzzleException.php [hunk 3]
```diff
@@ -21,9 +21,7 @@ class GuzzleException extends ConnectionException
     protected $_guzzleException;
 
     /**
-     * @param TransferException $guzzleException
-     * @param Request           $request
-     * @param Response|null     $response
+     * @param Request $request
```
Comment (thePanz) [inline lib/Elastica/Exception/Connection/GuzzleException.php]: Not sure why only `$request` is kept here :confused:

File: lib/Elastica/Multi/Search.php [hunk 4]
```diff
@@ -78,8 +74,7 @@ public function clearSearches(): self
     }
 
     /**
-     * @param BaseSearch $search
-     * @param string     $key
+     * @param string $key
```
Comment (thePanz) [inline lib/Elastica/Multi/Search.php]: This is not correct IMHO, as it should be `string|null`

File: lib/Elastica/Util.php [hunk 5]
```diff
@@ -198,8 +198,7 @@ public static function convertDate($date)
      *
      * Converts it to the lucene format, including the appropriate TimeZone
      *
-     * @param \DateTime $dateTime
-     * @param bool      $includeTimezone
+     * @param bool $includeTimezone
```
Comment (thePanz) [inline lib/Elastica/Util.php]: This should be added to the method signature, `bool $includeTimezone = true`

**Comment by thePanz** [comment]
@deguif could you address the couple of comments I left? just minor changes :)

**Comment by ruflin** [comment]
@thePanz Sorry, already merged. @deguif Any chance you could do a follow up PR with the above proposed changes?

Does adding php-cs-fixer to composer have any disadvantages? I assume there is a reason we don't have it here?

**Comment by deguif** [comment]
@ruflin I will do the follow up PR ;)

File: lib/Elastica/Multi/Search.php (hunk 4 above)
Comment (deguif) [reply lib/Elastica/Multi/Search.php]: maybe we should add `?string` in the type-hint of the method and remove the php-doc for the param?
Even better we could also use the `nullable_type_declaration_for_default_null_value` option for PHP-CS-FIXER to enforce the rule, WDYT?

Comment (thePanz) [reply lib/Elastica/Multi/Search.php]: In this case I believe it is an error on the docblock: we accept null, but the annotation not.
The annotation is more strict than the method signature.

I'd rather (1) remove the annotation (if we handle null values), or (2) check the implementation if null values are not accepted. If (2) the method signature is wrong, and then the annotation is superfluous :)

File: lib/Elastica/Exception/Connection/GuzzleException.php (hunk 3 above)
Comment (deguif) [reply lib/Elastica/Exception/Connection/GuzzleException.php]: For the same reason probably I explained below

File: lib/Elastica/Multi/Search.php (hunk 4 above)
Comment (deguif) [reply lib/Elastica/Multi/Search.php]: https://github.com/php/php-src/pull/3535

Comment (thePanz) [reply lib/Elastica/Multi/Search.php]: Didn't know about `nullable_type_declaration_for_default_null_value` and the discussion. LGTM to use it :+1: 

Anyway: is the method really accepting null values?

Comment (deguif) [reply lib/Elastica/Multi/Search.php]: Null values seem to be accepted, so I'm going to remove the annotation, as argument type-hint is sufficient

**Comment by deguif** [comment]
@ruflin @thePanz minor changes asked here are now addressed in #1706

**Pull request merged**